packages/flutter/lib/src/services/text_input.dart

```diff
@@ -16,47 +16,118 @@ export 'dart:ui' show TextAffinity;
 /// The type of information for which to optimize the text input control.
 ///
 /// On Android, behavior may vary across device and keyboard provider.
-enum TextInputType {
+///
+/// This class allows additional flags for some input types. For example,
+/// numeric input can specify whether it supports decimal numbers and/or
+/// signed numbers.
+class TextInputType {
+  const TextInputType._(this.type) : signed = null, decimal = null;
+
+  /// Optimize for textual information.
+  ///
+  /// Requests a numeric keyboard with additional settings.
+  /// The [signed] and [decimal] parameters are optional.
+  const TextInputType.numberWithOptions({
+    this.signed: false,
+    this.decimal: false,
+  }) : type = 2;
+
+  final int type;
+
+  /// The number is signed, allowing a positive or negative sign at the start.
+  /// This flag is only used for numeric input type.
+  final bool signed;
+
+  /// The number is decimal, allowing a decimal point to provide fractional.
+  /// This flag is only used for numeric input type.
+  final bool decimal;
+
   /// Optimize for textual information.
   ///
   /// Requests the default platform keyboard.
-  text,
+  static const TextInputType text = const TextInputType._(0);
 
   /// Optimize for multi-line textual information.
   ///
   /// Requests the default platform keyboard, but accepts newlines when the
   /// enter key is pressed. This is the input type used for all multi-line text
   /// fields.
-  multiline,
+  static const TextInputType multiline = const TextInputType._(1);
 
   /// Optimize for numerical information.
   ///
-  /// Requests a keyboard with ready access to the decimal point and number
-  /// keys.
-  number,
+  /// Requests a default keyboard with ready access to the number keys.
+  /// Additional settings, such as decimal point and/or positive/negative
+  /// signs, can be requested by using the [numberWithOptions] constructor.
```

Comment:
I believe you need `[new TextInputType.numberWithOptions]` to make that xref work.

Reply:
Just to confirm, using "new" here even though the actual usage is with "const"?